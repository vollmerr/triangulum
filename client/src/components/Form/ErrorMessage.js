import React from 'react';
import { Message } from 'semantic-ui-react';

// single message of all field and submission errors
const ErrorMessage = (props) => {
  const {
    errors,
    touched,
    status,
    schema,
  } = props;
  const list = [];

  Object.keys(schema).forEach((field) => {
    if (errors[field] && touched[field]) {
      list.push(errors[field]);
    }
  });

  if (status && status === 'submissionError') {
    list.push('Failed to Submit');
  }

  const messageProps = {
    list,
    error: true,
    header: 'There was some errors with your submission',
  };

  return list.length ? <Message {...messageProps} /> : null;
};

export default ErrorMessage;
