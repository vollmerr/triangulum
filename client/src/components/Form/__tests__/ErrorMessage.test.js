import React from 'react';
import { shallow } from 'enzyme';

import ErrorMessage from '../ErrorMessage';

const props = {
  errors: {
    errTouch: 'errTouch Error',
    errNoTouch: 'errNoTouch Error',
  },
  touched: {
    touch: true,
    errTouch: true,
    errNoTouch: false,
  },
  status: null,
  schema: {
    touch: {},
    errTouch: {},
    errNoTouch: {},
  },
};

let wrapper;
describe('FormikForm', () => {
  beforeEach(() => {
    wrapper = shallow(<ErrorMessage {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render if no errors', () => {
    wrapper.setProps({ errors: {} });
    expect(wrapper.isEmptyRender()).toBeTruthy();
  });

  it('should render a submission error status', () => {
    wrapper.setProps({ status: 'submissionError' });
    const list = wrapper.first().prop('list');
    expect(list).toContain('Failed to Submit');
  });
});
