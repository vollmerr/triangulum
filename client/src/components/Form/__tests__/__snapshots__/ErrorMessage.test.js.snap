// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikForm should render correctly 1`] = `
<Message
  error={true}
  header="There was some errors with your submission"
  list={
    Array [
      "errTouch Error",
    ]
  }
/>
`;
